### PR TITLE
Spring boot testing books categories

### PR DESCRIPTION
**Pull request by YaroslavYarmoshyk**

### CODE REVIEW

File: src/test/java/com/online/bookstore/controller/BookControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+        assertThat(actual.getTotalElements()).isEqualTo(15);
+        assertThat(actual.getNumberOfElements()).isEqualTo(3);
+        assertThat(actual.getContent())
+                .contains(THE_GREAT_GATSBY_BOOK)
+                .extracting(BookDto::title)
+                .containsExactly(
+                        "The Great Gatsby",
+                        "The Catcher in the Rye",
+                        "To the Lighthouse"
```
Comment (MentorMA) [inline src/test/java/com/online/bookstore/controller/BookControllerTest.java]: Strings you can save to constants

File: src/test/java/com/online/bookstore/controller/CategoryControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+        assertThat(actual.getNumberOfElements()).isEqualTo(0);
+        assertThat(actual).isEmpty();
+    }
+
+    @Test
+    @DisplayName("Test delete category by existing ID")
+    @WithMockUser(roles = "ADMIN")
+    @Rollback
+    void deleteCategory_WithExistingId_Success() throws Exception {
+        final MvcResult mvcResult = mockMvc.perform(delete(CATEGORIES_API + "/1"))
```
Comment (MentorMA) [inline src/test/java/com/online/bookstore/controller/CategoryControllerTest.java]: save "/1"to constant and reuse it

File: src/test/java/com/online/bookstore/utils/TestDataFactory.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+public final class TestDataFactory {
+    public static final BookDto THE_GREAT_GATSBY_BOOK = new BookDto(
+            4L,
+            "The Great Gatsby",
+            "F. Scott Fitzgerald",
+            "9780743273565",
+            BigDecimal.valueOf(12.50).setScale(2, RoundingMode.HALF_UP),
+            "A tragic love story set in the Jazz Age.",
+            "https://example.com/thegreatgatsby.jpg",
+            Set.of(1L, 7L)
```
Comment (MentorMA) [inline src/test/java/com/online/bookstore/utils/TestDataFactory.java:22]: You can create templates and reuse them

**Review by MentorMA — CHANGES_REQUESTED** [review]

File: src/test/java/com/online/bookstore/controller/CategoryControllerTest.java (hunk 2 above)
Comment (YaroslavYarmoshyk) [reply src/test/java/com/online/bookstore/controller/CategoryControllerTest.java]: sure, thanks

File: src/test/java/com/online/bookstore/controller/BookControllerTest.java (hunk 1 above)
Comment (YaroslavYarmoshyk) [reply src/test/java/com/online/bookstore/controller/BookControllerTest.java]: moved

File: src/test/java/com/online/bookstore/utils/TestDataFactory.java (hunk 3 above)
Comment (YaroslavYarmoshyk) [reply src/test/java/com/online/bookstore/utils/TestDataFactory.java:22]: could you please elaborate more on this?

**Review by koval-kn — APPROVED** [review]

**Pull request merged**